docs/source/developers/guide/step_by_step/testing.rst

```diff
@@ -62,29 +62,103 @@ In this section we outline steps needed for unit testing in Arrow.
+
+      .. code-block:: R
+
+         devtools::test()
+
+      in the R console. Alternatively, you can use
+
+      .. code:: console
+
+         $ make test
+
+      in the shell.
+
+      You can run the tests in a single test file you have open with
+
+      .. code-block:: R
+
+         devtools::test_active_file()
+
+      All tests are also run as part of our continuous integration (CI) pipelines.
+
+      **Good practice**
+
+      In general any change to source code needs to be accompanied by unit tests. All tests are expected to pass before a pull request is merged.
+
+      * Add functionality -> add unit tests
+      * Modify functionality -> update unit tests
+      * Solve a bug -> add unit test before solving it, which helps prove the bug and its fix
+      * Performance improvements should be reflected in benchmarks (which are also tests)
+      * An exception could be refactoring functionality that is fully covered by unit tests
+
+      **Testing helpers**
+
+      To complement the ``testthat`` functionality, the ``arrow`` R package has defined a series of specific utility functions (called helpers), such as:
+
+      * Expectations - these start with ``expect_`` and are used to compare objects
+            - for example, ``expect_altrep_roundtrip()`` compares the result
+            of a function ``fn`` run on a vector ``x`` with the result of the
+            same function run on the altrep version of ``x``. More generally,
+            expect_…_roundtrip() functions do …
+
+            .. TODO _fill in the blanks_
+            .. TODO
+
+            - Expect
+      * ``skip_`` - skips a unit test - think of them as acceptable fails. Situations in which we might want to skip unit tests:
+
+        - ``skip_if_r_version()`` - this is a specific ``arrow`` skip. For example, we use this to skip a unit test when the R version is 3.5.0 and below (``skip_if_r_version(“3.5.0”)``). You will likely see it used when the functionality we are testing depends on features introduced after version 3.5.0 of R (such as the alternative representation of vectors, Altrep, introduced in R 3.5.0, but with significant additions in subsequent releases). As part of our CI workflow we test against different versions of R and this is where this feature comes in.
+        - ``skip_if_not_available()`` - another specific {arrow} skip. Arrow (libarrow) has a series of additional features that can be switched on or off (but this needs to happen at build time). If a unit test depends on such a feature and this feature is not available (i.e. was not selected when libarrow was built) the test is skipped, as opposed to having a failed test.
+        - ``skip_if_offline()`` - will not run tests that require an internet connection
+        - ``skip_on_os()`` - for unit tests that are OS specific.
+
+      *Important*: Once the conditions for a ``skip_()`` statement is met, no other line of code in the same ``test_that()`` test block will get executed.
```

Comment:
```suggestion
      *Important*: Once the conditions for a ``skip_()`` statement is met, no other line of code in the same ``test_that()`` test block will get executed. If the skip is outside of a `test_that()` code block, it will skip the rest of the file.
```